### PR TITLE
Tables rework!

### PR DESCRIPTION
**Pull request by JPLachance**
+ Add support for the "check all" checkbox in a table
! Fix table row height
- Remove some useless nesting

### CODE REVIEW

**Comment by JPLachance** [comment]
screen shot 2016-07-06 at 4 26 09 pm
screen shot 2016-07-06 at 4 26 23 pm
screen shot 2016-07-06 at 4 26 35 pm

File: docs/_components/tables/01_table.html [hunk 1]
```diff
@@ -27,7 +27,11 @@
         </th>
         <th>Column 2</th>
         <th class="admin-sort">
-            Column 3
+            <label class="coveo-checkbox-label">
+                <input type="checkbox" class="coveo-checkbox">
+                <button type="button"></button>
+            </label>
+            <span class="cell-content">Column 3</span>
```
Comment (pochretien) [inline docs/_components/tables/01_table.html]: i see u use coveo- for the checkbox, we can use coveo-cell-content i think

File: scss/tables/fixed-header-table.scss [hunk 2]
```diff
@@ -30,17 +36,22 @@
...
-        &.selection .fixed-header-container {
-          padding-top: 4px;
-        }
-
         div.fixed-header-container {
           position: absolute;
           top: 0;
 
-          padding-top: 8px;
+          padding: (($table-header-inner-height - $table-header-line-height) / 2) 0;
```
Comment (pochretien) [inline scss/tables/fixed-header-table.scss]: we can create a variable with the calc padding

File: scss/tables/table.scss [hunk 3]
```diff
@@ -22,104 +20,82 @@ table {
...
+    line-height: $table-header-line-height;
 
     background-color: transparent;
+  }
 
-    th {
-      line-height: normal;
-      white-space: nowrap;
+  th {
+    padding: (($table-header-inner-height - $table-header-line-height) / 2) $table-cell-left-right-padding;
```
Comment (pochretien) [inline scss/tables/table.scss]: reuse the variable created before ( in comment )

File: scss/tables/table.scss [hunk 4]
```diff
@@ -279,13 +127,11 @@ table {
       text-align: center;
     }
 
-    &.admin-focus {
-      td {
-        background-color: #f5f9fd;
+    &.admin-focus td {
+      background-color: #f5f9fd;
```
Comment (pochretien) [inline scss/tables/table.scss]: var color

File: scss/tables/table.scss [hunk 5]
```diff
@@ -279,13 +127,11 @@ table {
...
 
-        &.admin-focus-cell {
-          background-color: #ebf3fb;
-        }
+      &.admin-focus-cell {
+        background-color: #ebf3fb;
```
Comment (pochretien) [inline scss/tables/table.scss]: same here, var color

File: scss/tables/table.scss [hunk 6]
```diff
@@ -300,6 +146,51 @@ table {
     }
   }
 
+  td {
+    padding: (($table-row-inner-height - $table-row-line-height) / 2) $table-cell-left-right-padding;
```
Comment (pochretien) [inline scss/tables/table.scss]: create variable for the calc

File: docs/_components/tables/01_table.html (hunk 1 above)
Comment (JPLachance) [inline docs/_components/tables/01_table.html]: See https://github.com/coveo/vapor/issues/136

File: scss/tables/table.scss (hunk 3 above)
Comment (wfortin) [inline scss/tables/table.scss]: +1

File: scss/tables/table.scss [hunk 7]
```diff
@@ -300,6 +146,51 @@ table {
...
+
+      cursor: pointer;
+    }
+
+    .btn-icon, .dropdown-toggle {
+      cursor: default;
+      opacity: 0.3;
+
+      &.enabled:active:hover {
+        border-radius: 4px;
```
Comment (pochretien) [inline scss/tables/table.scss]: why 4px and not 2px ?

File: scss/tables/table.scss (hunk 5 above)
Comment (wfortin) [inline scss/tables/table.scss]: Color var here also.

**Comment by wfortin** [comment]
🚢  it

File: scss/tables/fixed-header-table.scss (hunk 2 above)
Comment (JPLachance) [inline scss/tables/fixed-header-table.scss]: Done.

File: scss/tables/table.scss (hunk 3 above)
Comment (JPLachance) [inline scss/tables/table.scss]: Done!

File: scss/tables/table.scss (hunk 4 above)
Comment (JPLachance) [inline scss/tables/table.scss]: Unused classes... removing them.

File: scss/tables/table.scss (hunk 5 above)
Comment (JPLachance) [inline scss/tables/table.scss]: Same, unused, deleting.

File: scss/tables/table.scss (hunk 7 above)
Comment (JPLachance) [inline scss/tables/table.scss]: Changed.

File: scss/tables/table.scss (hunk 6 above)
Comment (JPLachance) [inline scss/tables/table.scss]: Done!

**Pull request merged**